src/downloader/headers/fetch_request_stage.rs

```diff
@@ -58,6 +58,13 @@ impl FetchRequestStage {
             self.pending_count()
         );
         self.request_pending()?;
+
+        // in case of SendQueueFull, await for extra capacity
+        if self.pending_count() > 0 {
+            let queue_space_future = self.sentry.read().send_message_queue_space_wait();
```

Comment:
why not in one line?

Reply:
This is because I need to release the lock before awaiting the result. While awaiting other stages run in parallel, and might want to use the sentry object.

I've made this playground to test:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6425344be497c65c44e4bb507e6352b8

Notice how in the 2nd case "2" temporary is dropped before the "newline" (before `make("4")` call), but in the first case it is held during `make("4")`.

Reply:
Interesting. However, "2" is dropped in both before 4. Only 3 is held until the end in the second case. 
Hence, I'd assume that the lock would also be released, in both cases. But I'm not 100% sure, so please don't put so much thought on this :).

Reply:
> However, "2" is dropped in both before 4.

Yes, that's the point. In this case "2" corresponds to .read() (RwLockGuard temporary), "3" is the future, "4" is await. I want "2" be dropped before "4" (RwLockGuard unlock before await).

Reply:
But wouldn't it then happen in both cases, if it's `2` that's being dropped right away?

Reply:
So if you look at this playground:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6425344be497c65c44e4bb507e6352b8

in case 1 ("oneline") 2 is held, and until make("4"), then it drops 4, then it drops 2
in case 2 ("split") 2 is dropped before we make("4")

Reply:
Got it! Now I follow. Thanks :)